app/models/manageiq/providers/lenovo/physical_infra_manager/operations/ansible_sender.rb

```diff
@@ -0,0 +1,98 @@
+#
+# Mixin to run ansible operations
+#
+module ManageIQ::Providers::Lenovo::PhysicalInfraManager::Operations::AnsibleSender
+  extend ActiveSupport::Concern
+
+  #
+  # Executes ansible resources inside +root_dir+
+  #
+  # @param args [Hash] - The args to run the ansible resource.
+  # @option args [String] 'playbook_name' - The name of the playbook to be executed.
+  #         If this param is filled a yaml file that matches with the specified name
+  #         will be searched inside +playbooks_dir+.
+  #         In cases where the playbook is not in +playbooks_dir+ root, please, specify
+  #         the relative path to it.
+  #         Example:
+  #           run_ansible({'playbook_name' => 'config/firmware_update_all_dev'})
+  #
+  # @option args [String] 'role_name' - The name of the role to be executed.
+  #        If this param is filled it will search the role inside +roles_dir+.
+  #        Example:
+  #           run_ansible({'role_name' => 'lenovo.lxca-inventory'})
+  #
+  # @option args [Hash] vars - The set of vars needed to execute the ansible resource
+  #
+  # @return Ansible::Runner::Response
+  #
+  # @see #root_dir
+  # @see #playbooks_dir
+  # @see #roles_dir
+  #
+  def run_ansible(args = {})
+    playbook_name, role_name, vars = load_params(args)
+    if playbook_name.present?
+      Ansible::Runner.run(
+        {},
+        ansible_default_vars.merge(vars),
+        playbooks_dir.join(playbook_name =~ /\.yml/ ? playbook_name : "#{playbook_name}.yml")
```

Comment:
Similar point here, why not just enforce that you need the full path?  We own all of the callers so it isn't for end-user ease-of-use

Reply:
I agree with you here, just joining `playbook_name` is enough